src/wfcomman.c

```diff
@@ -1021,52 +1021,61 @@ AppCommandProc(register DWORD id)
-		   DWORD cchEnv;
-		   TCHAR szToRun[MAXPATHLEN];
-		   LPTSTR szDir;
+   {
+      BOOL bRunAs;
+      BOOL bUseCmd;
+      BOOL bDir;
+      DWORD cchEnv;
+      TCHAR szToRun[MAXPATHLEN];
+      LPTSTR szDir;
 
 #define ConEmuParamFormat TEXT(" -Single -Dir \"%s\"")
-           TCHAR szParams[MAXPATHLEN + COUNTOF(ConEmuParamFormat)];
+      TCHAR szParams[MAXPATHLEN + COUNTOF(ConEmuParamFormat)];
 
-			szDir = GetSelection(1|4|16, &bDir);
-			if (!bDir && szDir)
-				StripFilespec(szDir);
-	
-		   bRunAs = GetKeyState(VK_SHIFT) < 0;
-
-		   // check if conemu exists: %ProgramFiles%\ConEmu\ConEmu64.exe
-		   bUseCmd = TRUE;
-		   cchEnv = GetEnvironmentVariable(TEXT("ProgramFiles"), szToRun, MAXPATHLEN);
-		   if (cchEnv != 0) {
-			   if (lstrcmpi(szToRun + cchEnv - 6, TEXT(" (x86)")) == 0) {
-				   szToRun[cchEnv - 6] = TEXT('\0');
-			   }
-               // NOTE: assume ProgramFiles directory and "\\ConEmu\\ConEmu64.exe" never exceed MAXPATHLEN
-               lstrcat(szToRun, TEXT("\\ConEmu\\ConEmu64.exe"));
-			   if (PathFileExists(szToRun)) {
-				   wsprintf(szParams, ConEmuParamFormat, szDir);
-				   bUseCmd = FALSE;
-			   }
-		   }
+      szDir = GetSelection(1 | 4 | 16, &bDir);
+      if (!bDir && szDir)
+         StripFilespec(szDir);
 
-		   // use cmd.exe if ConEmu doesn't exist or we are running admin mode
-		   if (bUseCmd || bRunAs) {
-               // NOTE: assume system directory and "\\cmd.exe" never exceed MAXPATHLEN
-               if (GetSystemDirectory(szToRun, MAXPATHLEN) != 0)
-                    lstrcat(szToRun, TEXT("\\cmd.exe"));
-               else
-			        lstrcpy(szToRun, TEXT("cmd.exe"));
-			   szParams[0] = TEXT('\0');
-		   }
+      bRunAs = GetKeyState(VK_SHIFT) < 0;
 
-		   ret = ExecProgram(szToRun, szParams, szDir, FALSE, bRunAs);
-		   LocalFree(szDir);
-		}
-	   break;
+      // check if conemu exists: %ProgramFiles%\ConEmu\ConEmu64.exe
+      bUseCmd = TRUE;
+      cchEnv = GetEnvironmentVariable(TEXT("ProgramFiles"), szToRun, MAXPATHLEN);
+      if (cchEnv != 0) {
+         if (lstrcmpi(szToRun + cchEnv - 6, TEXT(" (x86)")) == 0) {
+            szToRun[cchEnv - 6] = TEXT('\0');
+         }
+         // NOTE: assume ProgramFiles directory and "\\ConEmu\\ConEmu64.exe" never exceed MAXPATHLEN
+         lstrcat(szToRun, TEXT("\\ConEmu\\ConEmu64.exe"));
+         if (PathFileExists(szToRun)) {
+            wsprintf(szParams, ConEmuParamFormat, szDir);
+            bUseCmd = FALSE;
+         }
+      }
+
+      // use cmd.exe if ConEmu doesn't exist or we are running admin mode
+      if (bUseCmd) {
+         // NOTE: assume system directory and "\\cmd.exe" never exceed MAXPATHLEN
+         if (GetSystemDirectory(szToRun, MAXPATHLEN) != 0)
+            lstrcat(szToRun, TEXT("\\cmd.exe"));
+         else
+            lstrcpy(szToRun, TEXT("cmd.exe"));
+
+         if (bRunAs) {
+            // Windows >= 8 ignores the 5th parameter of ShellExecute aka 'lpDirectory' when elevating, 

```

Comment:
I don't think this part is true; I tried on Windows 7 and see the same behavior.  As far as I know Vista also has it.  (The issue/problem was if an unprivileged account can plant a DLL or similar in a directory, tricking an elevated program into invoking that code.  Forcing system32 was a mitigation that works because system32 can't be modified by an unprivileged user.)

Reply:
This is odd. I also tried it on my W7 and cmd opens as admin in the given location. But maybe my W7 is too much a developer setup. With W10 it definitely ignores this when elevating.
I don't know where I read this, but it stated there that with introduction of UAC the fifth parameter is ignored.

Anyhow I have deleted the comment at all

Reply:
Just some theories - does your Win7 have UAC enabled, running a split-token Administrator?  (Mine had UAC disabled and I had to enable it and create a new user account to test this PR.)